experanto/interpolators.py

```diff
@@ -39,18 +37,40 @@ def interpolate(self, times: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
     def __contains__(self, times: np.ndarray):
         return np.any(self.valid_times(times))
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        self.close()
+
     @staticmethod
     def create(root_folder: str, cache_data: bool = False, **kwargs) -> "Interpolator":
         with open(Path(root_folder) / "meta.yml", "r") as file:
             meta_data = yaml.load(file, Loader=yaml.SafeLoader)
         modality = meta_data.get("modality")
-        class_name = modality.capitalize() + "Interpolator"
-        assert class_name in globals(), f"Unknown modality: {modality}"
-        return globals()[class_name](root_folder, cache_data, **kwargs)
+
+        if modality == "sequence":
+            if meta_data.get("phase_shift_per_signal", False):
+                return PhaseShiftedSequenceInterpolator(
+                    root_folder, cache_data, **kwargs
+                )
+            else:
+                return SequenceInterpolator(root_folder, cache_data, **kwargs)
+        elif modality == "screen":
+            return ScreenInterpolator(root_folder, cache_data, **kwargs)
+        else:
+            raise ValueError(
+                f"There is no interpolator for {modality}. Please use 'sequence' or 'screen' as modality."
+            )
 
     def valid_times(self, times: np.ndarray) -> np.ndarray:
         return self.valid_interval.intersect(times)
 
+    def close(self):
+        ...
+        # generally, nothing to do
+        # can be overwritten to close any open files or resources
+
```

Comment:
why do we need this? 
like its not called anywhere in practice and I do not see why would it but I might be missing sth
(sorry for being so picky, I just try that we do not overload experanto with software for the sake of software)

Reply:
ok, I see, thanks a lot for the explanation.
I guess it's another case of python garbage collector not being efficient enough. Let me finish going through the tests and decide if we keep it or not:)

Reply:
ok, lets keep the closing as I do not now the better elegant solution for the tests right now

Reply:
@reneburghardt  should we resolve this one as well?